File: ch05/number.zig
const std = @import("std");

const Number = enum(u8) { Zero, One, Two, _ };

pub fn main() !void {
    var num = Number.Zero;
    std.debug.print("{any} => {d}\n", .{ num, @intFromEnum(num) });
    // Increment to One
    num = @enumFromInt(@intFromEnum(num) + 1);
    std.debug.print("{any} => {d}\n", .{ num, @intFromEnum(num) });
    // Increment to Two
    num = @enumFromInt(@intFromEnum(num) + 1);
    std.debug.print("{any} => {d}\n", .{ num, @intFromEnum(num) });

    num = @enumFromInt(@intFromEnum(num) + 1);
    std.debug.print("{any} => {d}\n", .{ num, @intFromEnum(num) });
}

File: ch05/number.c
#include<stdio.h>

typedef enum{
    Zero, 
    One, 
    Two, 
}Number;


const char* numberStr(Number num){
    switch (num)
    {
    case Zero: 
        return "Number.Zero";
        break;
    case One: 
        return "Number.One";
        break;
    case Two: 
        return "Number.Two";
        break;
    default:
        break;
    }
}

int main(){
    // Start num at zero 
    Number num = Zero;
    printf("%s => %d\n", numberStr(num), num);
    // Increment to go to One
    ++num;
    printf("%s => %d\n", numberStr(num), num);
    // Increment to go to Two
    ++num;
    printf("%s => %d\n", numberStr(num), num);

    return 0;
}